### ISSUE TITLE
Unable to start service com.guichaguri.trackplayer.service.MusicService@4185a5

### ISSUE THREAD

**Issue by tyhour** [doublesymmetry/react-native-track-player#1532]
**Describe the Bug**
App force close after use ` await TrackPlayer.setupPlayer();` I think it want problem with TargetSDKVersion 31.

Because it show like:
`Caused by: java.lang.IllegalArgumentException: com.we_read_app: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
    Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.`

**Environment Info:**
```gradle
ext  {
        buildToolsVersion = "31.0.0"
        minSdkVersion = 21
        compileSdkVersion = 31
        targetSdkVersion = 31

        if (System.properties['os.arch'] == "aarch64") {
            // For M1 Users we need to use the NDK 24 which added support for aarch64
            ndkVersion = "24.0.8215888"
        } else if (Os.isFamily(Os.FAMILY_WINDOWS)) {
            // For Android Users, we need to use NDK 23, otherwise the build will
            // fail due to paths longer than the OS limit
            ndkVersion = "23.1.7779620"
        } else {
            // Otherwise we default to the side-by-side NDK version from AGP.
            ndkVersion = "21.4.7075529"
        }
    }
```

**Package Install**
```json
"dependencies": {
    "react": "17.0.2",
    "react-native": "0.68.1",
    "react-native-track-player": "^2.1.3",
  }
```
**Thanks for help**
Please help me for some solution.

**Comment by jspizziri** [comment]
Please search old issues before filing a new bug (this is one that's been reported multiple times).

The short version is that android 12 & SDK 31 are supported in  2.2.0-rc3 and we will not be backporting a fix to lower versions (anything at or before 2.1).

https://github.com/doublesymmetry/react-native-track-player/issues/1193

**Comment by jspizziri** [comment]
You'll either need to wait for [this issue](https://github.com/doublesymmetry/react-native-track-player/issues/1418) to be resolved or contribute the fix.

**Comment by tyhour** [comment]
> You'll either need to wait for [this issue](https://github.com/doublesymmetry/react-native-track-player/issues/1418) to be resolved or contribute the fix.

Ok thanks.

**Comment by Strate** [comment]
To fix issue on 2.1.* branch apply this:

```
diff --git a/node_modules/react-native-track-player/android/build.gradle b/node_modules/react-native-track-player/android/build.gradle
index a241d94..b17a4a6 100644
--- a/node_modules/react-native-track-player/android/build.gradle
+++ b/node_modules/react-native-track-player/android/build.gradle
@@ -88,7 +88,7 @@ dependencies {
 
     // Make sure we're using androidx
     implementation "androidx.core:core:1.3.0-rc01"
-    implementation "androidx.media:media:1.1.0"
+    implementation "androidx.media:media:1.6.0"
     implementation "androidx.localbroadcastmanager:localbroadcastmanager:1.1.0-alpha01"
     implementation "com.github.bumptech.glide:glide:4.7.1"
 }
 diff --git a/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java b/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java
index 42dc212..72c42e4 100644
--- a/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java
+++ b/node_modules/react-native-track-player/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java
@@ -81,7 +81,7 @@ public class MetadataManager {
         openApp.setAction(Intent.ACTION_VIEW);
         openApp.setData(Uri.parse("trackplayer://notification.click"));
 
-        builder.setContentIntent(PendingIntent.getActivity(context, 0, openApp, PendingIntent.FLAG_CANCEL_CURRENT));
+        builder.setContentIntent(PendingIntent.getActivity(context, 0, openApp, PendingIntent.FLAG_CANCEL_CURRENT | (Build.VERSION.SDK_INT >= 31 ? PendingIntent.FLAG_MUTABLE : 0)));
 
         builder.setSmallIcon(R.drawable.play);
         builder.setCategory(NotificationCompat.CATEGORY_TRANSPORT);
```

**Comment by phindell** [comment]
Also to fix the issue on 2.1.*, you can upgrade to **2.1.4** : it includes the fix of @Strate, in a version made by devpolo ! See [version comparison](https://github.com/doublesymmetry/react-native-track-player/compare/v2.1.3...v2.1.4), the [last commit](https://github.com/doublesymmetry/react-native-track-player/pull/1555)

(Context: we decided to do that as a short-term fix with my team, as we had too many issues on iOS by bumping to 3.* - but still we need immediate Android 12 support for Google Play Store validation!)
